### PR TITLE
[YANG] Align to authentication check with load_minigraph to cover more scenarios

### PR DESCRIPTION
**Pull request by wen587**
#### Why I did it
This will cover more cases for those only have authentication enabled with tacacs but not authorization enabled. Also, make it algin with the check in load_minigraph
##### Work item tracking
- Microsoft ADO **(number only)**: 26732148

#### How I did it
Change to authentication
#### How to verify it
unit test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang [hunk 1]
```diff
@@ -43,8 +43,8 @@ module sonic-system-aaa {
                     default "local";
                 }
 
-                must 'not(./type = "authorization" and contains(./login, "tacacs+") and not(/tacacs:sonic-system-tacacs/tacacs:TACPLUS/tacacs:global/tacacs:passkey))' {
-                    error-message "Authorization with 'tacacs+' is not allowed when passkey not exists.";
+                must 'not(./type = "authentication" and contains(./login, "tacacs+") and not(/tacacs:sonic-system-tacacs/tacacs:TACPLUS/tacacs:global/tacacs:passkey))' {
+                    error-message "Authentication with 'tacacs+' is not allowed when passkey not exists.";
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-models/sonic-system-aaa.yang:47]: Should we have 2 `must` or just one `must`?

Comment (wen587) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang:47]: For our case, authentication is enough. It is a superset of authorization

**Review by liuh-80 — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang:47]: Let me be more clear, if
1. authentication is local
2. authorization is tacacs+
3. no passkey

Is it a valid config or not?

Comment (wen587) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang:47]: It is not valid in Azure. Becuase Azure requires a passkey to connect to tacacs server.

Comment (qiluo-msft) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang:47]: Offline discussed, the conclusion is that this is valid config.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**